### PR TITLE
share button updates

### PR DESCRIPTION
**Pull request by osahyoun**

### CODE REVIEW

File: app/workers/share_analytics_updater.rb [hunk 1]
```diff
@@ -17,16 +24,27 @@ def update_collection
...
-        puts $!.message
-        next
-      end
+      Aws::SQS::Client.new.send_message({
+        queue_url:    ENV['SQS_ENDPOINT'],
+        message_body: {
+          type: 'update_share',
+          share_id: button.id
+        }
+      }.to_json)
```
Comment (osahyoun) [inline app/workers/share_analytics_updater.rb]: @Tuuleh with this the cron would spit out messages into the queue, so creating one job for each share.

**Comment by Tuuleh** [comment]
I've merged this onto `fix-cron` and will be working on it there

**Comment by osahyoun** [comment]
This PR fixes the concurrency problems we were having on our worker. I've put a couple hundred jobs on the queue for share progress data updates and the worker has comfortably handled the load:

image

Log output starts from Feb 24 15:06:18 ak-worker-staging

The worker's health flatlines (in a good way) right after my deployment.

image

**Comment by osahyoun** [comment]
@Tuuleh and others, I'd love for this to be reviewed. Would be good to get it into master before I head off for the day

**Comment by NealJMD** [comment]
this looks good to me!

File: app/controllers/message_handler_controller.rb [hunk 2]
```diff
@@ -2,20 +2,12 @@ class MessageHandlerController < ApplicationController
...
-    # as is the standard for the Worker Tier in Elastic Beanstalk.
     response = QueueListener.new.perform(nil, params)
-
-    # Tell SQS everything went hunky dory and we can delete the message.
     render json: response, status: 200
   end
 
   def dispatch_shareprogress_update
-    ShareAnalyticsUpdater.update_shares
+    ShareAnalyticsUpdater::EnqueueJobs.run
```
Comment (Tuuleh) [inline app/controllers/message_handler_controller.rb]: Didn't you add the class method so you could call `ShareAnalyticsUpdater.enqueue_jobs`?

Comment (osahyoun) [inline app/controllers/message_handler_controller.rb]: yes, so I did. Seems a little redundant now. I think I'll drop `enqueue_jobs`

File: circle.yml [hunk 3]
```diff
@@ -32,7 +32,7 @@ deployment:
       - docker push soutech/champaign-ak-processor
       - ./deploy.sh $CIRCLE_SHA1 'champaign' 'ak-worker-production' 'logs3.papertrailapp.com:44107'
   staging:
-    branch: ak_params
+    branch: testing
```
Comment (Tuuleh) [inline circle.yml]: change back to development

File: spec/lib/ak_page_creator_spec.rb [hunk 4]
```diff
@@ -1,61 +0,0 @@
-require 'rails_helper'
```
Comment (Tuuleh) [inline spec/lib/ak_page_creator_spec.rb]: Why do these specs fly out? Looks unrelated to this PR

Comment (osahyoun) [inline spec/lib/ak_page_creator_spec.rb]: They're old specs that were replicating recently made request specs.

**Comment by Tuuleh** [comment]
:100: :balloon: :cake: Waiting for this to build and test on CCI, will merge after.
